Source/DataSources/KmlDataSource.js

```diff
@@ -1947,20 +1945,52 @@ define([
         Tour : processUnsupported
     };
 
+    var promisifiedFeatureTypes = {
```

Comment:
There's no need for both `promisifiedFeatureTypes` and `featureTypes`, I would just use `featureTypes` and update the functions.